### ISSUE TITLE
General repo maintenance

### ISSUE THREAD

**Issue by sujaypatil96** [linkml/linkml-model-enrichment#56]
Some practices we can choose to follow in order to tidy the repository contents:

* Rename the repository as documented in issue #55. Also, rename the source folder in the repo root with the same name.
* Within the linkml_model_enrichment folder in the repo root, we can add an `__init__.py` file and turn it into a package. Similarly, we can add an `__init__.py` within each of the sub folders, making each of them sub packages.
  * We need the source folder to be a package so we can reference the modules for the CLI using the dot notation, as we do in the pyproject.toml file
* Is the `local` folder necessary?
* I assume `pre_poetry` has contents which were present before the current version of the repo? This can be removed because we can always get to that version by going back through the commit history.
* Do we need the `target` folder? Unless there are paths in the source CLIs that default to this path, I think we can safely remove this and allow the users to configure the target folders themselves?
* Separate dependencies listed in pyproject.toml file into mandatory dependencies and dev dependencies.
  * Remove dependencies that are imported circularly. For ex., we don't need to have linkml-runtime as a dependency if we already have linkml listed
* Remove requirements.txt and setup.py

CC: @cmungall

**Comment by turbomam** [comment]
Great, @sujaypatil96 . I'll review this in the next day or two.

**Comment by turbomam** [comment]
@sujaypatil96 can you please make sure that #47 gets merged in? I can do that if you tell me a good time to do it.

**Comment by turbomam** [comment]
My contributions for enum annotation need major refactoring, but I don't mean to say you should do that.

**Comment by sierra-moxon** [comment]
@sujaypatil96  - these still seem like a good list of things to do, is the idea that it's just a very low priority? :) If not going to be completed in a reasonable amount of time, maybe we unassign and close?

**Comment by sujaypatil96** [comment]
@sierra-moxon: yup, this was just low priority at the time. I can put this issue in our backlogs, add a low priority label, and unassign myself. A new person coming into the repo can look and it and take it up if he/she has the time.

**Comment by amc-corey-cox** [comment]
Closing — most items on this checklist have been completed (repo rename, setup.py removal, pypi publishing, dependency cleanup). Please open specific issues for any remaining items.
